libraries/AP_Scripting/AP_Scripting_CANSensor.h

```diff
@@ -0,0 +1,73 @@
+/*
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+/*
+  Scripting CANSensor class, for easy scripting CAN support
+ */
+ 
+#pragma once
+
+#include <AP_CANManager/AP_CANSensor.h>
+
+#if HAL_MAX_CAN_PROTOCOL_DRIVERS
+class ScriptingCANBuffer;
+class ScriptingCANSensor : public CANSensor {
+public:
+
+    ScriptingCANSensor():CANSensor("Script", AP_CANManager::Driver_Type::Driver_Type_Scripting) {};
```

Comment:
```suggestion
    ScriptingCANSensor():CANSensor("Script") {
        register_driver(AP_CANManager::Driver_Type::Driver_Type_Scripting);
    }
```

@IamPete1  please rebase on top of this PR https://github.com/ArduPilot/ardupilot/pull/18089 . Add above changes and you should be able to remove the added delay.